lib/make-badge.js

```diff
@@ -119,16 +117,10 @@ function makeBadge (measurer, {
     text = text.map(value => value.toUpperCase());
   }
 
-  if (colorscheme) {
-    let pickedColorscheme = definedColorschemes[colorscheme];
-    if (!pickedColorscheme) {
-      pickedColorscheme = definedColorschemes.red;
-    }
-    colorA = colorA || pickedColorscheme.colorA;
-    colorB = colorB || pickedColorscheme.colorB;
-  }
-  if (!cssColor.test(colorA)) { colorA = undefined; }
-  if (!cssColor.test(colorB)) { colorB = undefined; }
+  colorA = colorA || colorscheme || undefined;
+  colorB = colorB || colorscheme || undefined;
+  colorA = definedColorschemes[colorA] !== undefined ? definedColorschemes[colorA].colorA || undefined : validColor(colorA) ? colorA : undefined;
```

Comment:
In previous implementation `pickedColorscheme = definedColorschemes.red;` was responsible for overriding invalid `colorscheme` value. Now we will get an `undefined` instead of a `red`. This way svg plastic badge has brightgreen color (https://github.com/badges/shields/blob/master/templates/flat-template.svg?short_path=164000c#L13) instead of `red`.

Reply:
I'm not sure this would be too much of an issue as someone has just as much chance of mistyping the `label` or any other part of the badge as the `colorscheme`.
And when mistyped would fall back to the badges preferred default color (or css3 named color if one exist) or hopefully be caught during review/test.

There is a couple of ways this could be fixed to return `red` though:

could change the lines to be:
```javascript
const errorColor = definedColorschemes.red.colorB;
...
  colorA = definedColorschemes[colorA] !== undefined ? definedColorschemes[colorA].colorA || errorColor : validColor(colorA) ? colorA : errorColor;
  colorB = definedColorschemes[colorB] !== undefined ? definedColorschemes[colorB].colorB || errorColor : validColor(colorB) ? colorB : errorColor;
```
which would always return `#e05d44` when given an invalid color,
essentially removing the need for default colors in the badge template.

or could use a getter/setter in `badge-data.js`:

```javascript
function setBadgeColor(badgeData, color) {
  if (isHexColor(color)) {
    badgeData.colorB = '#' + color;
    delete badgeData._colorscheme;
  } else if (colorschemes[color] !== undefined){
    badgeData.colorscheme = color;
    delete badgeData.colorB;
  } else {
    badgeData.colorB = color;
    delete badgeData._colorscheme;
  }
  return badgeData;
}

function makeBadgeData(defaultLabel, overrides) {
  return {
    text: [makeLabel(defaultLabel, overrides), 'n/a'],
    _colorscheme: 'lightgrey',
    get colorscheme(){
      return this._colorscheme || undefined;
    },
    set colorscheme(color){
      this._colorscheme = colorschemes[color] !== undefined ? color : 'red';
    },
    template: overrides.style,
    logo: makeLogo(undefined, overrides),
    logoWidth: +overrides.logoWidth,
    links: toArray(overrides.link),
    colorA: makeColor(overrides.colorA),
    colorB: makeColor(overrides.colorB),
  };
}
```

Reply:
In addition to the above (1st option) we could also add #1522 
```json
{
  "brightgreen":    { "colorA": "#555", "colorB": "#4c1" },
  "green":          { "colorA": "#555", "colorB": "#97ca00" },
  "yellow":         { "colorA": "#555", "colorB": "#dfb317" },
  "yellowgreen":    { "colorA": "#555", "colorB": "#a4a61d" },
  "orange":         { "colorA": "#555", "colorB": "#fe7d37" },
  "red":            { "colorA": "#555", "colorB": "#e05d44" },
  "blue":           { "colorA": "#555", "colorB": "#007ec6" },
  "grey":           { "colorA": "#555", "colorB": "#555" },
  "gray":           { "colorA": "#555", "colorB": "#555" },
  "lightgrey":      { "colorA": "#555", "colorB": "#9f9f9f" },
  "lightgray":      { "colorA": "#555", "colorB": "#9f9f9f" },

  "error":          { "colorA": "#555", "colorB": "#e05d44" },
  "notice":         { "colorA": "#555", "colorB": "#fe7d37" },
  "success":        { "colorA": "#555", "colorB": "#4c1" },
  "info":           { "colorA": "#555", "colorB": "#007ec6" },
  "default":        { "colorA": "#555", "colorB": "#9f9f9f" }
}
```

Reply:
1. I don't know if `lightgrey` is better than `red`. For me we can stay with `red`. 
2. I really like the idea of removing default colors from badge templates (can be done in a separate PR).
3. How about this option?
Add this coloscheme (`red` as `default`):
```json
"default":        { "colorA": "#555", "colorB": "#e05d44" }
```
plus
```js
const detaultColor = definedColorschemes.default;
...
colorA = definedColorschemes[colorA] !== undefined ? definedColorschemes[colorA].colorA || detaultColor.colorA : validColor(colorA) ? colorA : detaultColor.colorA;
colorB = definedColorschemes[colorB] !== undefined ? definedColorschemes[colorB].colorB || detaultColor.colorB : validColor(colorB) ? colorB : detaultColor.colorB;
```
without adding `colorA` in all color schemes (we don't need them, right)? 

And maybe you're right that overriding invalid colorcheme is not an such important issue.

Reply:
1. :+1: 
2. Should be fine as long as each `colorscheme` has set `colorA`.
3. Think this may be the best option for now, will do this.